### PR TITLE
Feature/upgrade openid package

### PR DESCRIPTION
**Pull request by jekuaitk**
Upgraded from itk-dev/openid-connect 1.0.0 to 2.1.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,31 +136,32 @@ class TestAuthenticator extends OpenIdLoginAuthenticator
...
     private $entityManager;
 
-    public function __construct(EntityManagerInterface $entityManager, SessionInterface $session, UrlGeneratorInterface $router)
+    public function __construct(EntityManagerInterface $entityManager, SessionInterface $session, UrlGeneratorInterface $router, OpenIdConfigurationProvider $provider)
     {
         $this->router = $router;
         $this->entityManager = $entityManager;
-        parent::__construct($session);
+        // Set leeway directly or configure via .env (must be positive)
+        parent::__construct($provider, $session, $leeway);
```
Comment (turegjorup) [inline README.md:143]: I know this is just an example in the readme, but `Undefined variable '$leeway' ` - consider adding it as a constructor argument.

Also `ExampleAuthenticator` is probably a better name than `TestAuthenticator`

File: README.md [hunk 2]
```diff
@@ -136,31 +136,32 @@ class TestAuthenticator extends OpenIdLoginAuthenticator
...
-        // Update/set names here
+        // Update/set user properties
         $user->setName($name);
         $user->setEmail($email);
 
-        // persist and flush user to database
-        // If no change persist will recognize this
+        // Persist and flush user to database
+        // If no change, persist will recognize this
         $this->entityManager->persist($user);
```
Comment (turegjorup) [inline README.md]: It's only necessary to call `persist()` on objects that are not handled by the `EntityManager`, should be moved inside the `if (null === $user) {` conditional

File: src/Security/OpenIdLoginAuthenticator.php [hunk 3]
```diff
@@ -29,27 +40,38 @@ public function supports(Request $request)
...
 
+    /**
+     * @throws ValidationException
+     */
     public function getCredentials(Request $request)
     {
-        // Make sure state and oauth2sate are the same
+        // Make sure state and oauth2state are the same
         if ($request->query->get('state') !== $this->session->get('oauth2state')) {
             $this->session->remove('oauth2state');
```
Comment (turegjorup) [inline src/Security/OpenIdLoginAuthenticator.php]: How is `oauth2state` removed if `state` match?

**Review by turegjorup — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jekuaitk) [reply README.md:143]: Added `$leeway` to constructer and gave example binding via `.env` and `services.yaml`.

Updated name to ExampleAuthenticator.

File: README.md (hunk 2 above)
Comment (jekuaitk) [reply README.md]: Moved persist into conditional.

File: src/Security/OpenIdLoginAuthenticator.php (hunk 3 above)
Comment (jekuaitk) [reply src/Security/OpenIdLoginAuthenticator.php]: Made sure to remove `oauth2state`.

**Review by turegjorup — APPROVED** [review]

**Pull request merged**